src/main/java/dk/bec/polonez/reservationsystem/service/OfferService.java

```diff
@@ -35,36 +41,79 @@ public OfferService(OfferRepository offerRepository, FeatureRepository featureRe
     }
 
     public OfferDto getById(long id) {
-        Offer offer = offerRepository.getById(id);
-        return modelMapper.map(offer, OfferDto.class);
+        Optional<Offer> offer = offerRepository.findById(id);
+        return modelMapper.map(offer.orElseThrow(
+                () -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Offer with id=" + id + " not found")), OfferDto.class);
     }
 
     public OfferDto addOffer(CreateOfferDto offerDto) {
-        Offer offer = modelMapper.map(offerDto, Offer.class);
-        Offer savedOffer = offerRepository.save(offer);
-        return modelMapper.map(savedOffer, OfferDto.class);
+        if (authService.isPoLoggedIn() || authService.isAdminLoggedIn()) {
+            Offer offer = modelMapper.map(offerDto, Offer.class);
+            offer.setOwner(authService.getCurrentUser());
+            Offer savedOffer = offerRepository.save(offer);
+            return modelMapper.map(savedOffer, OfferDto.class);
+        } else {
+            throw new AuthenticationException(HttpStatus.FORBIDDEN, "Access denied, log as Place Owner or System Admin");
+        }
     }
 
     public OfferDto updateOffer(long id, CreateOfferDto offerDto) {
-        Offer offer = offerRepository.getById(id);
-        modelMapper.map(offerDto, offer);
-        Offer updatedOffer = offerRepository.save(offer);
-        return modelMapper.map(updatedOffer, OfferDto.class);
+        Optional<Offer> optionalOffer = offerRepository.findById(id);
+        if (authService.isPoLoggedIn() || authService.isAdminLoggedIn()) {
```

Comment:
lepiej by bylo to sprawdzac privilege, czyli ze kazda rola, ma przypisane liste privilege
i np. jest privilege CAN_UPDATE_OFFER 

dzieki temu, jesli cos sie zmieni i np. admin ma nie miec mozliwosci updatowac offerty, to wtedy zabieracie mu ten privilege z bazy i juz tego nie moze robic, a przy obecnym rozwiazaniu muscie zmienic ten kod ;)